src/basilisp/lang/interfaces.py

```diff
@@ -374,6 +374,18 @@ def pop(self: Self) -> Self:
         raise NotImplementedError()
 
 
+class IReduce:
+    @abstractmethod
+    def reduce(self, f, init):
+        raise NotImplementedError()
+
+
+class IReduceKV:
+    @abstractmethod
+    def reduce_kv(self, f, init):
+        raise NotImplementedError()
```

Comment:
Please use `__slots__ = ()` on each. This helps constrain memory usage of Basilisp objects.

Reply:
Done.